src/k8s/pkg/k8sd/types/cluster_config.go

```diff
@@ -64,14 +68,13 @@ func DefaultClusterConfig() ClusterConfig {
 			Port: 9000,
 		},
 	}
-}
 
-// ClusterConfigFromBootstrapConfig extracts the cluster config parts from the BootstrapConfig
-// and maps them to a ClusterConfig.
-func ClusterConfigFromBootstrapConfig(b *apiv1.BootstrapConfig) ClusterConfig {
```

Comment:
Let's keep this function here, this should give us a ClusterConfig seeded from the BootstrapConfig.

Also, two CIDRs (one IPv4, one IPv6) may be specified, so if we want to validate it, use https://github.com/canonical/k8s-snap/blob/7e04e1b16dbb252052e56c237f3449d85eca7212/src/k8s/pkg/component/network.go#L18-L38 for inspration.

```
func ClusterConfigFromBootstrapConfig(b *apiv1.BootstrapConfig) (ClusterConfig, error) {
  if _, _, err := net.ParseCIDR(b.ClusterCIDR); err != nil {
    return ClusterConfig{}, fmt.Errorf("cluster CIDR is not valid: %w", err)
  }
  return ClusterConfig{ .... }, nil
}
```

Reply:
Thanks Angelos,
the validating CIDR is updated 👍🏼 
I felt like the validation part might be better placed after we get the bootstrap config interactively. Let me know your thoughts on that.

Reply:
yeah, probably best to have a `(c *ClusterConfig) Validate() error`